crates/fervid_core/src/options.rs

```diff
@@ -0,0 +1,51 @@
+use std::collections::HashMap;
+
+/// Asset URL tag configuration
+/// Example: { "img": ["src"], "link": ["href"] }
+pub type AssetURLTagConfig = HashMap<String, Vec<String>>;
+
+#[derive(Debug, Clone)]
+pub struct AssetURLOptions {
+    /// Base path for rewriting URLs
+    pub base: Option<String>,
+    /// Whether to process absolute URLs
+    pub include_absolute: bool,
+    /// Tag-specific configuration
+    pub tags: Option<AssetURLTagConfig>,
```

Comment:
I think this needs to be non-option, because it is `fervid_core`.

The assumption is: `fervid_core` gets "clean" structures into it. Otherwise every crate which depends on it will need to `unwrap` with some kind of default values. `fervid_core` can provide defaults as you already did.

```suggestion
    pub tags: AssetURLTagConfig,
```